src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -859,10 +859,15 @@ void activity_classifier::init_train(
     for (size_t i = 1; i < gpu_names.size(); ++i) {
       gpu_names_string += ", " + gpu_names[i];
     }
-    logprogress_stream << "Using "
-                       << (gpu_names.size() > 1 ? "GPUs" : "GPU")
-                       << " to create model ("
-                       << gpu_names_string << ")";
+    if (gpu_names_string.find("/") != std::string::npos) {
```

Comment:
Could you add a comment explaining the logic here, since it's pretty fragile? Maybe a TODO to find something more robust in future